### PR TITLE
Fixed #364

### LINKED ISSUE #364: Reorganizing the mods list crashes the server wrapper while in themed mode.
**Torch Version:** 1.3.1.100 **SE Version:** 1.19.3.103 (Experimental Mode)

### Expected Behavior
You can drag and drop the mods in the mod list up or down freely no matter the theme.

### Observed Behavior
When dragging a mod downward in the list it crashes while in a theme other than Torch Theme. Although when dragging a mod upward in the list it works just fine.

### Steps to Reproduce
1. Start with a fresh install
2. Set the theme to Light Theme or Dark Theme and it can be the animated ones.
3. Add at least two mods in the mods list if not added already
4. Drag a mod upward in the list (It shouldn't crash)
5. Drag a mod downward in the list.
6. Crash
7. Repeat steps 4 5 with the Torch theme, and it should work fine.

### Other Information
I did some debugging with commit `5035fa3` and I have found that when dragging upward the values for the parameter `values` of the function `Torch.Server.Views.Converters.ModToListIdConverter.Convert(object[] values, Type targetType, object parameter, CultureInfo culture)` the array has two indexes (0 and 1) and they are correct and correspond to the objects that the function tries to cast them as, and when dragging downward the two indexes are not correctly cast as they do not have the correct data to cast, the data that is in both of them is `_name` as a key, and `undefined` as a value.

Here is a log file of the crash https://pastebin.com/arvK5y8g

### PR DESCRIPTION
**Pull request by thakyZ**
The cause of this issue was because of these steps.
1. The list couldn't be dragged downward thus returning null to ModToIdConverter
2. The list was unable to insert the value at an index that was changed.
3. The length of the index when dragging becomes 1 if the list is 2
4. Since the length was 1 it would try to insert the value at index 1 (length of 2)

This has been tested by me, someone should try it as well.

### CODE REVIEW

**Comment by thakyZ** [comment]
I had fixed the issue somewhat in d651ea1 but when I did some more testing the issue sort of still persisted if you were going to try to bug it out, which wasn't hard to do, so I made commit 7f37aca to try to fix that, and it seems to work.

File: Torch.Server/Views/Converters/ModToIdConverter.cs [hunk 1]
```diff
@@ -32,9 +32,16 @@ public object Convert(object[] values, Type targetType, object parameter, Cultur
         {
             //if (targetType != typeof(int))
             //    throw new NotSupportedException("ModToIdConverter can only convert mods into int values or vise versa!");
-            var mod = (ModItemInfo) values[0];
-            var theModList = (MtObservableList<ModItemInfo>) values[1];
-            return theModList.IndexOf(mod);
+            if (values[0].GetType() != typeof(ModItemInfo) || values[1].GetType() != typeof(MtObservableList<ModItemInfo>))
```
Comment (rexxar-tc) [inline Torch.Server/Views/Converters/ModToIdConverter.cs]: This is not how you check types in C#. Use the `is` keyword.

File: Torch.Server/Views/ModListControl.xaml.cs [hunk 2]
```diff
@@ -205,9 +205,9 @@ private void UserControl_MouseMove(object sender, MouseEventArgs e)
             {
                 _hasOrderChanged = true;
                 var modList = (MtObservableList<ModItemInfo>)DataContext;
-                //modList.Move(modList.IndexOf(_draggedMod), modList.IndexOf(targetMod));
-                modList.RemoveAt(modList.IndexOf(_draggedMod));
-                modList.Insert(modList.IndexOf(targetMod), _draggedMod);
+                modList.Move(modList.IndexOf(targetMod), _draggedMod);
```
Comment (rexxar-tc) [inline Torch.Server/Views/ModListControl.xaml.cs]: You should ensure that your index is valid before passing it on. If `targetMod` is null, you'll get interesting exceptions.

File: Torch.Server/Views/Converters/ModToIdConverter.cs (hunk 1 above)
Comment (thakyZ) [reply Torch.Server/Views/Converters/ModToIdConverter.cs]: I have modified it now to fix the checking of types

File: Torch.Server/Views/ModListControl.xaml.cs (hunk 2 above)
Comment (thakyZ) [reply Torch.Server/Views/ModListControl.xaml.cs]: I have already checked if `targetMod` is null, in fact on line L204 there is code that was already implemented to check if `targetMod` was not null.

File: Torch.Server/Views/Converters/ModToIdConverter.cs [hunk 3]
```diff
@@ -32,9 +32,14 @@ public object Convert(object[] values, Type targetType, object parameter, Cultur
         {
             //if (targetType != typeof(int))
             //    throw new NotSupportedException("ModToIdConverter can only convert mods into int values or vise versa!");
-            var mod = (ModItemInfo) values[0];
-            var theModList = (MtObservableList<ModItemInfo>) values[1];
-            return theModList.IndexOf(mod);
+            if (values[0] is ModItemInfo mod || values[1] is MtObservableList<ModItemInfo> modList)
```
Comment (rexxar-tc) [inline Torch.Server/Views/Converters/ModToIdConverter.cs]: You inverted the logic from the last change. The `||` should now be `&&`

Comment (thakyZ) [reply Torch.Server/Views/Converters/ModToIdConverter.cs]: Oh right ok thank you.

Comment (thakyZ) [reply Torch.Server/Views/Converters/ModToIdConverter.cs]: Okay I have fixed in in 
d7196b8

**Pull request merged**